### PR TITLE
[Platforms] Add s3 credentials mounter (modifier)

### PR DESCRIPTION
**Pull request by yaronha**

### CODE REVIEW

File: mlrun/platforms/other.py [hunk 1]
```diff
@@ -165,3 +166,75 @@ def _mount_hostpath(task):
...
 
     return _mount_hostpath
+
+
+def mount_s3(
+    aws_access_key="",
+    aws_secret_key="",
+    secret_name=None,
+    endpoint_url=None,
+    aws_region=None,
```
Comment (theSaarco) [inline mlrun/platforms/other.py:176]: Why is this needed? I don't recall we're using it anywhere, and in general S3 access is not region-governed (each bucket is in a region, but you don't specify a region when accessing it)

File: mlrun/platforms/other.py [hunk 2]
```diff
@@ -165,3 +166,75 @@ def _mount_hostpath(task):
...
+                        secret_key_ref=k8s_client.V1SecretKeySelector(
+                            name=secret_name, key="AWS_SECRET_ACCESS_KEY"
+                        )
+                    ),
+                )
+            )
+
+        else:
+            return container_op.add_env_variable(
+                k8s_client.V1EnvVar(name="AWS_ACCESS_KEY_ID", value=_access_key)
```
Comment (theSaarco) [inline mlrun/platforms/other.py:235]: I'm not comfortable with enabling this non-secure flow which will expose AWS creds in the function spec, as well as in the UI/sdk. At a minimum this should show a visible warning that notifies user about the potential risks (also in the docstring). 
I was thinking that this could be a utility function, such that if the user provides visible creds, it will automatically plant them (well, probably just the `AWS_SECRET_ACCESS_KEY`) in a project secret, saving the user the need to do that.

**Review by theSaarco — CHANGES_REQUESTED** [review]
We need to discuss the cleartext option - I don't think it should be endorsed by this utility function. In general, MLRun is currently providing a confusing set of options to pass credentials, and we need to also consider how we add better docs to guide the users with it - when do you use project secrets, when (and how) you need to create a k8s secret and mount it, what the difference is between `set_env_from_secret`, `mount_secret`, this new function and so on.

File: mlrun/platforms/other.py (hunk 1 above)
Comment (yaronha) [reply mlrun/platforms/other.py:176]: i copied it from the s3 modifier in KFP, i assume it has usage

File: mlrun/platforms/other.py (hunk 2 above)
Comment (yaronha) [reply mlrun/platforms/other.py:235]: we have the same in mount_v3io, i'm ok with adding a warning, and there is a way to use `secret_name`, we shouldn't make modifiers to complex. AFAIK the S3 datastore driver is already looking into the project secret

Comment (yaronha) [reply mlrun/platforms/other.py]: added log warning

**Review by theSaarco — APPROVED** [review]
LGTM. 
Note that prefix support is partial - the datastore code itself does not look for env variables with any prefix, there will need to be another PR which implements that.

File: mlrun/platforms/other.py [hunk 3]
```diff
@@ -165,3 +167,87 @@ def _mount_hostpath(task):
...
+
+
+def mount_s3(
+    secret_name=None,
+    aws_access_key="",
+    aws_secret_key="",
+    endpoint_url=None,
+    prefix="",
+    aws_region=None,
+):
```
Comment (Hedingber) [inline mlrun/platforms/other.py:179]: I honestly it would make more sense and will be more user friendly (and will be easier to maintain for us) to have a simple `with_s3_creds` function, instead of this modifier thing

**Review by Hedingber — CHANGES_REQUESTED** [review]

**Comment by yaronha** [comment]
We need this for the auto mount, in the kit, regardless we can think of with_..

**Review by Hedingber — APPROVED** [review]
Talked offline, we're adding the mount option for users who would like one creds for all jobs, in which we would like to leverage our auto mount mechanism, which supports only modifier functions
